charts/argo/templates/server-ingress.yaml

```diff
@@ -23,24 +25,37 @@ metadata:
     {{- end }}
 spec:
   rules:
-    {{- range .Values.server.ingress.hosts }}
-    - host: {{ . }}
+  {{- if .Values.server.ingress.hosts }}
+  {{- range $host := .Values.server.ingress.hosts }}
+    - host: {{ $host }}
       http:
         paths:
-          {{- if $.Values.server.ingress.paths }}
-          {{- range $.Values.server.ingress.paths }}
-          - backend:
-              serviceName: {{ .serviceName }}
-              servicePort: {{ .servicePort }}
-          {{- end }}
-          {{- end }}
-          - backend:
+  {{- if $extraPaths }}
+  {{- toYaml $extraPaths | nindent 10 }}
+  {{- end }}
+  {{- range $p := $paths }}
+          - path: {{ $p }}
+            backend:
```

Comment:
Why do you reformat the leading spaces? I find it really confusing when every templating section is left-sided.
Can we use the existing indentation?